pkg/util/spanlogger/spanlogger.go

```diff
@@ -34,6 +39,9 @@ func New(ctx context.Context, method string, kvps ...interface{}) (*SpanLogger,
 // retrieved with FromContext or FromContextWithFallback.
 func NewWithLogger(ctx context.Context, l log.Logger, method string, kvps ...interface{}) (*SpanLogger, context.Context) {
 	span, ctx := opentracing.StartSpanFromContext(ctx, method)
+	if ids, _ := tenant.TenantIDs(ctx); len(ids) > 0 {
```

Comment:
This is going to add tenant tag to every span created via this method. That seems unnecessary.

Reply:
It's true that's unnecessary, but is that a bad thing?

Reply:
Alternatively, we can add functional options to a `NewWithLoggerWithOptions()` function an conditionally enable it only when the option is specified. What do you think?

Reply:
> Alternatively, we can add functional options to a `NewWithLoggerWithOptions()` function an conditionally enable it only when the option is specified. What do you think?

You've convinced me that it's ok. I would keep it as is.

Reply:
> Alternatively, we can add functional options to a NewWithLoggerWithOptions() function an conditionally enable it only when the option is specified. What do you think?

it was the first implementation when I added separate method to create a span with tenant tag. however, all existing span creations had to be revised and some of them had to be replaced with a new method. otherwise tenant_ids won't be defined for some spans when we investigate something on tracing dashboard. 
I found that current implementation the easiest way and much more safe because when somebody add a new code he won't forget to create a span with tenant tag.

> This is going to add tenant tag to every span created via this method. That seems unnecessary.
unfortunately, yes. if it's critical we can think how to add this tag only if `tenant_federation.enabled` is true.

Reply:
> @pstibrany
> unfortunately, yes. if it's critical we can think how to add this tag only if `tenant_federation.enabled` is true.

After some thinking and looking at our existing spans, I think it's fine... one extra tag per span is no big deal, and one can immediately see tenant when checking the trace.

Reply:
@pstibrany fully agree with you ;) thanks for your review

Reply:
It's always good to have conversations! 💪